src/NHibernate.Test/Linq/ByMethod/JoinTests.cs

```diff
@@ -103,6 +99,98 @@ public void LeftJoinExtensionMethodWithMultipleKeyProperties()
 			}
 		}
 
+		[Test]
+		public void LeftJoinExtensionMethodWithOuterReferenceInWhereClauseOnly()
+		{
+			using (var sqlSpy = new SqlLogSpy())
+			{
+				var animals = db.Animals
+							   .LeftJoin(
+								   db.Mammals,
+								   x => x.Id,
+								   x => x.Id,
+								   (animal, mammal) => new { animal, mammal })
+							   .Where(x => x.mammal.SerialNumber.StartsWith("9"))
+							   .Select(x => new { SerialNumber = x.animal.SerialNumber })
+							   .ToList();
+
+				var sql = sqlSpy.GetWholeLog();
+				Assert.That(animals.Count, Is.EqualTo(1));
+				Assert.That(GetTotalOccurrences(sql, "left outer join"), Is.EqualTo(1));
+			}
+		}
+
+		[KnownBug("GH-2379")]
```

Comment:
It's not 2379. It's #2739 

And what about https://github.com/nhibernate/nhibernate-core/issues/2672. Is it related/same?

I believe should be reproduced using the following slightly modified test case:
```C#
//GH-2672
[Test]
public void LeftJoinExtensionMethodWithOuterReferenceInWhereClauseOnlyCount()
{
	using (var sqlSpy = new SqlLogSpy())
	{
		var count= db.Animals
						.LeftJoin(
							db.Mammals,
							x => x.Id,
							x => x.Id,
							(animal, mammal) => new {animal, mammal})
						.Where(x => x.mammal.SerialNumber.StartsWith("9"))
						.Select(x => new { x.mammal.Pregnant})
							//.Select(x => new {SerialNumber = x.animal.SerialNumber, x.animal.Description})
						//.ToList()
						.Count();

		var sql = sqlSpy.GetWholeLog();
		Assert.That(count, Is.EqualTo(1));
		Assert.That(GetTotalOccurrences(sql, "left outer join"), Is.EqualTo(1));
	}
}
```

Reply:
Fixed the bug number and added a fix for #2672.

Reply:
Fix doesn't look related.. Am I right? Maybe better to fix in separate PR? Just a nit